services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/validation/ApplicationValidator.kt

```diff
@@ -19,8 +19,7 @@ class ApplicationValidator : ConstraintValidator<ValidApplication, BaseApplicati
         val result =
             when (val data = application.applicationData) {
                 is CableReportApplicationData -> validate { data.validateForErrors() }
-                is ExcavationNotificationApplicationData ->
-                    TODO("Excavation notification not implemented")
+                is ExcavationNotificationData -> validate { data.validateForErrors() }
```

Comment:
Do we support excavation notifications somewhere where this validation is useful?

Reply:
Should I leave the actual validation code or delete that too? It will be under hakemus package in the future and you are atm doing that.

Reply:
I think we'll still need support for excavation notices for `ensureValidForSend()`, so the validation code shouldn't be removed.